static/js/poll_modal.js

```diff
@@ -0,0 +1,76 @@
+import $ from "jquery";
+import {Sortable} from "sortablejs";
+
+import render_poll_modal_option from "../templates/poll_modal_option.hbs";
+
+export function update_option_delete_button_visibility($container, display_flag) {
+    const no_of_option_row = $container.find(".option-row").length;
+
+    // When there is only one option row, we hide its delete button
+    if (no_of_option_row === 1) {
+        // display_flag indicates whether another row will be added
+        // next, and in that case, we should show the delete button
+        if (display_flag === true) {
+            $container.find(".option-row .delete-option").show();
+        } else {
+            $container.find(".option-row .delete-option").hide();
+        }
+    }
```

Comment:
This logic feels confusing; could we instead just compute the number of rows we expect to be present, and then show/hide based on that?

I'm also not sure why it's correct to do nothing if the current number of rows is 0 or 2.

Reply:
A simpler non-nested logic based on only the final expected row count is possible, but would do more work than necessary. The crux here is that only when the current row count is 1, we may need to change the show / hide status of the 1 row depending on if another will be added right after (`display_flag`). This is also the reason we only do something if the row count is 1, and not 0, or 2 or anything else. (Also, count can never really be 0, since this function is never called in that state)

Reply:
We don't expect there to be more than a few dozen rows even for very large polls, so I'm not sure I understand the performance concern?

Reply:
Hm yes, the performance concerns aren't relevant here then, it's only linear in the row count. I'll update the code to the simpler code proposed above.